src/main/groovy/edu/oregonstate/mist/recommendations/resources/RecommendationResource.groovy

```diff
@@ -27,46 +28,103 @@ class RecommendationResource extends Resource {
 
     @GET
     @Produces(MediaType.APPLICATION_JSON)
-    public List<Recommendation> getRecommendations(
-            @QueryParam('by') @NotEmpty String by,
-            @QueryParam('student_type') @NotEmpty String studentType,
-            @QueryParam('province') @NotEmpty String province,
-            @QueryParam('batch') @NotEmpty Integer batch,
-            @QueryParam('lower_limit') @NotEmpty Integer lowerLimit,
-            @QueryParam('upper_limit') @NotEmpty Integer upperLimit,
+    public Response getRecommendations(
+            @QueryParam('by') String by,
+            @QueryParam('student_type') String studentType,
+            @QueryParam('province') String province,
+            @QueryParam('batch') Integer batch,
+            @QueryParam('lower_limit') Integer lowerLimit,
+            @QueryParam('upper_limit') Integer upperLimit,
             @QueryParam('year') Optional<Integer> year,
             @QueryParam('major') Optional<String> major,
             @QueryParam('language') Optional<String> language,
             @QueryParam('page_size') Optional<Integer> pageSize,
             @QueryParam('page_number') Optional<Integer> pageNum) {
 
+        if (!(by && studentType && province && batch && upperLimit && lowerLimit != null)) {
+            return badRequest("Some of the required query parameters are not provided!").build()
```

Comment:
I believe you can directly use `if (by && studentType && province && batch && upperLimit && lowerLimit)` here.